Mathlib/RingTheory/Derivation/Basic.lean

```diff
@@ -40,8 +40,9 @@ equality. We also require that `D 1 = 0`. See `Derivation.mk'` for a constructor
 assumption from the Leibniz rule when `M` is cancellative.
 
 TODO: update this when bimodules are defined. -/
-structure Derivation (R : Type*) (A : Type*) [CommSemiring R] [CommSemiring A] [Algebra R A]
-    (M : Type*) [AddCommMonoid M] [Module A M] [Module R M] extends A →ₗ[R] M where
+structure Derivation (R : Type*) (A : Type*) (M : Type*)
```

Comment:
I guess `A` can even be a `Semiring`, at least in this definition.

Reply:
That wouldn't make any sense without https://github.com/leanprover-community/mathlib/pull/18936

Reply:
Ah yes, `Module` still wants a commutative ring.

Reply:
That's not quite true; the generalization you suggest is accepted by Lean, but it is not a reasonable definition for the non-commutative case.